### DISCUSSION TITLE
Static and Dynamic VLAN Pool with range support

### DISCUSSION THREAD

**Discussion by niamtokik** [netbox-community/netbox#6102] [Q&A] [answered]
Hi everyone,

We need  to have a way to deal with static and dynamic VLAN allocation in our platform, but, it seems this feature is not implemented on Netbox. A similar feature requested was made in January 2020 (https://github.com/netbox-community/netbox/issues/3251), but was closed.

- Is this feature still impossible to create/implement on Netbox?
- Do you have more information about the reason why the model can't be altered? 

Thanks

**Comment by candlerb** [comment, marked as answer]
The reason was given [here](https://github.com/netbox-community/netbox/issues/3251#issuecomment-504128328).  Netbox models VLANs as individual items, not as ranges.  It would be difficult to model all possible use cases, such as QinQ.

You could pre-populate VLANs with status 'reserved', and change them to 'active' when allocated.  Alternatively, Netbox now supports [plugins](https://netbox.readthedocs.io/en/stable/plugins/) which would allow you to implement your own data model for VLAN pools.

**Reply by niamtokik** [reply]
Thanks for the quick answer. I will take a look on the plugins support. Do you know if a similar feature is already existing in plugins?
